### DISCUSSION TITLE
F´ Release v3.2.0

### DISCUSSION THREAD

**Discussion by thomas-bc** [nasa/fprime#1977] [Announcements]
## Release Notes
F´ v.3.2.0 comes with FPP, Tooling and GDS improvements. The underlying typing system has been improved to allow logical types.

## Breaking Changes
* Authors of CMake toolchains should now supply a PlatformTypes.h header according to the [Numerical Type design](https://nasa.github.io/fprime/Design/numerical-types.html). Linux example provided here: [cmake/platform/types/PlatformTypes.h](https://github.com/nasa/fprime/blob/devel/cmake/platform/types/PlatformTypes.h).
* LinuxSerialDriver has been renamed LinuxUartDriver and has been updated to support the ByteStreamDriver model. Users will need to update their ports.
* Users should consider adding ComQueue and ComAdapter to their downlink chain. 
* CAssert.hpp renamed CAssert.h

## Major Changes
* GDS now supports complex types, command arguments and parameters.
* Sanitization has been added to Unit Tests.
* FPP now directly generates data types and ports.
* `fprime-util new` now supports generating new projects (as well as deployments and components). Users are recommended to use this new functionality moving forward.
* Getting Started tutorial has been reworked and leverages the `fprime-util new` functionality.

## New Contributors
* @legounicycler made their first contribution in https://github.com/nasa/fprime/pull/1558
* @SMorettini made their first contribution in https://github.com/nasa/fprime/pull/1572
* @Anirban166 made their first contribution in https://github.com/nasa/fprime/pull/1589
* @Schweinepriester made their first contribution in https://github.com/nasa/fprime/pull/1591
* @SimonWaldherr made their first contribution in https://github.com/nasa/fprime/pull/1614
* @tiffany1618 made their first contribution in https://github.com/nasa/fprime/pull/1629
* @redixin made their first contribution in https://github.com/nasa/fprime/pull/1660
* @howjmay made their first contribution in https://github.com/nasa/fprime/pull/1668
* @fchapoton made their first contribution in https://github.com/nasa/fprime/pull/1644
* @codeflight1 made their first contribution in https://github.com/nasa/fprime/pull/1699
* @EbenezerA99 made their first contribution in https://github.com/nasa/fprime/pull/1653
* @arizvi786 made their first contribution in https://github.com/nasa/fprime/pull/1718
* @gazayas made their first contribution in https://github.com/nasa/fprime/pull/1736
* @thomas-bc made their first contribution in https://github.com/nasa/fprime/pull/1739
* @kevinthegreat1 made their first contribution in https://github.com/nasa/fprime/pull/1777
* @LMPS97 made their first contribution in https://github.com/nasa/fprime/pull/1822
* @dmitri-mcguckin made their first contribution in https://github.com/nasa/fprime/pull/1854
* @usamah1 made their first contribution in https://github.com/nasa/fprime/pull/1908
* @ivanperez-keera made their first contribution in https://github.com/nasa/fprime/pull/1928
* @pcrosemurgy made their first contribution in https://github.com/nasa/fprime/pull/1935

**Full Changelog**: https://github.com/nasa/fprime/compare/v3.1.1...v3.2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/nasa/fprime/releases/tag/v3.2.0'>F´ Release v3.2.0</a>.</em>

**Comment by LeStarch** [comment]
Wooooo!

**Comment by ThibFrgsGmz** [comment]
Awesome, we have a better homepage with the logo now!  😃 
    cc. @Anirban166 @astroesteban
